code/modules/mob/living/carbon/human/life.dm

```diff
@@ -85,16 +85,17 @@
 
 
 /mob/living/carbon/human/calculate_affecting_pressure(pressure)
-	var/chest_covered = FALSE
-	var/head_covered = FALSE
-	for(var/obj/item/clothing/equipped in get_equipped_items())
-		if((equipped.body_parts_covered & CHEST) && (equipped.clothing_flags & STOPSPRESSUREDAMAGE))
-			chest_covered = TRUE
-		if((equipped.body_parts_covered & HEAD) && (equipped.clothing_flags & STOPSPRESSUREDAMAGE))
-			head_covered = TRUE
-
-	if(chest_covered && head_covered)
-		return ONE_ATMOSPHERE
+	if (wear_suit && head && isclothing(wear_suit) && isclothing(head))
+		var/obj/item/clothing/CS = wear_suit
+		var/obj/item/clothing/CH = head
+		if ((CS.clothing_flags & STOPSPRESSUREDAMAGE) && (CH.clothing_flags & STOPSPRESSUREDAMAGE))
+			return ONE_ATMOSPHERE
+		if ((CS.clothing_flags & STOPSLOWPRESSUREDMG) && (CH.clothing_flags & STOPSLOWPRESSUREDMG))
+			if  (pressure < ONE_ATMOSPHERE)
+				return ONE_ATMOSPHERE
+		if ((CS.clothing_flags & STOPSHIGHPRESSUREDMG) && (CH.clothing_flags & STOPSHIGHPRESSUREDMG))
+			if (pressure > ONE_ATMOSPHERE)
+				return ONE_ATMOSPHERE
```

Comment:
```suggestion
	var/pressure_flags = STOPSPRESSUREDAMAGE | STOPSLOWPRESSUREDMG | STOPSHIGHPRESSUREDMG
	var/chest_protection = 0
	var/head_protection = chest_protection
	for(var/obj/item/clothing/equipped in get_equipped_items())
		if(equipped.body_parts_covered & CHEST)
			chest_protection |= equipped.clothing_flags & pressure_flags
		if(equipped.body_parts_covered & HEAD)
			head_protection |= equipped.clothing_flags & pressure_flags
	
	if(chest_protection & head_protection & STOPSPRESSUREDAMAGE)
		return ONE_ATMOSPHERE
	if(chest_protection & head_protection & STOPSLOWPRESSUREDMG)
		if (pressure < ONE_ATMOSPHERE)
			return ONE_ATMOSPHERE
	if(chest_protection & head_protection & STOPSHIGHPRESSUREDMG)
		if (pressure < ONE_ATMOSPHERE)
			return ONE_ATMOSPHERE
```

Untested code that might help preserve previous behavior in regards to pressure-proof items outside of suit/head slots (for example xenobio space-proofing potion on jumpsuit)

Reply:
Whoops, high pressure damage has `<` instead of `>` when checking atmosphere